File: chat-backend/src/main/java/com/smallchat/backend/room/framework/web/ChatController.java
package com.smallchat.backend.room.framework.web;

import com.smallchat.backend.global.utils.JwtProvider;
import com.smallchat.backend.room.application.usecase.ChatListUseCase;
import com.smallchat.backend.room.framework.web.dto.ChatListDto;
import lombok.AllArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.UUID;

@AllArgsConstructor
@RestController
@RequestMapping("/api/v2/rooms")
public class ChatController {
    private final ChatListUseCase chatListUseCase;
    private final JwtProvider jwtProvider;

    @GetMapping("/{roomId}/chats")
    public ResponseEntity<ChatListDto.Response> getChatHistory(@RequestHeader("Authorization") String authorization, @PathVariable String roomId) {
        jwtProvider.parseFromBearer(authorization);
        ChatListDto.Response chatList = chatListUseCase.getChattingList(UUID.fromString(roomId));
        return ResponseEntity.ok(chatList);
    }
}

File: chat-backend/src/main/java/com/smallchat/backend/chat/interfaces/web/ChatController.kt
package com.smallchat.backend.chat.interfaces.web

import com.smallchat.backend.chat.application.CreateChatUseCase
import com.smallchat.backend.chat.interfaces.web.dto.CreateChatDto
import com.smallchat.backend.global.infrastructure.web.CurrentUser
import com.smallchat.backend.global.utils.AuthenticatedUser
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v2/chats")
class ChatController(
    private val createChatUseCase: CreateChatUseCase,
) {
    @PostMapping
    fun createChat(
        @CurrentUser authenticatedUser: AuthenticatedUser,
        @RequestBody request: CreateChatDto.Request
    ): CreateChatDto.Response {
        val chatId = createChatUseCase.execute(authenticatedUser, request)
        return CreateChatDto.Response(chatId)
    }
}
